website.py

```diff
@@ -222,6 +245,24 @@
 updateDarkMode();
 darkModeCheckbox.addEventListener("change", updateDarkMode);
 
+function updateFunctions() {
+    var functionSelect = document.getElementsByName("functionselect")[0];
+    var prevValue = functionSelect.value;
+    var options = "<option value='all'>all functions</option>";
+    for (var line of sourceEl.value.split("\\n")) {
+        //var match = line.match(/^\\w*glabel\\w*([A-Za-z0-9_]+)/)
+        var match = line.match(/^\\s*glabel\\s+([A-Za-z0-9_]+)/)
+        if (match) {
+            var name = match[1];
+            options += "<option value='" + name + "' " + (name == prevValue ? "selected" : "") + ">" + name + "</option>";
```

Comment:
can you use document.createElement here instead, to avoid the XSS? (not that it matters in practice)